pkg/api/types.go

```diff
@@ -138,6 +138,21 @@ type ObjectMeta struct {
 	// objects.  Annotation keys have the same formatting restrictions as Label keys. See the
 	// comments on Labels for details.
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of objects depended by this object. If ALL objects in the list have
+	// been deleted, this object will be garbage collected.
+	OwnerReferences []ObjectReference `json:"ownerReferences,omitempty"`
+
+	// Must be empty before the object is deleted from the registry. The map key
+	// is an ad hoc, explanatory name (e.g., "orphaning finalizer"), the value
+	// is an identifier for the responsible component that will remove the entry
+	// from the map. If the ObjectMeta.DeletionTimestamp is non-nil, entry in
+	// this map can only be removed.
+	Finailzers map[string]string `json:"finalizers,omitempty"`
+
+	// Should the dependents be orphaned when this object is deleted. Default to
+	// false.
+	OrphanDependents *bool `json:"orphanDependents,omitempty"`
```

Comment:
We don't need both an orphan setting AND the finalizer list. lemme go over your proposal, we should pick the design.

Additionally, orphan should have a default value that depends on the object and api version-- default false for new objects, false for existing ones.